lib/results.js

```diff
@@ -104,7 +104,10 @@ Results.prototype._watch = function (t) {
         self.count ++;
 
         if (res.ok) self.pass ++
-        else self.fail ++
+        else {
+            self.emit("fail");
+            self.fail ++;
```

Comment:
could the sync emission fail? it seems like we should increment `self.fail` first.